openmls/src/group/mls_group/create_commit.rs

```diff
@@ -124,18 +143,19 @@ impl MlsGroup {
 
         let joiner_secret = JoinerSecret::new(
             backend,
-            provisional_tree.commit_secret(),
+            commit_secret_option.as_ref(),
             self.epoch_secrets()
                 .init_secret()
                 .ok_or(MlsGroupError::InitSecretNotFound)?,
         )?;
 
         // Create group secrets for later use, so we can afterwards consume the
         // `joiner_secret`.
-        let plaintext_secrets = PlaintextSecret::new(
+        let plaintext_secrets = PlaintextSecret::from_plain_update_path(
+            &diff,
             &joiner_secret,
             apply_proposals_values.invitation_list,
-            &provisional_tree,
+            plain_path_option.as_deref(),
```

Comment:
I can change the signature to use `impl`, but I think I still have to use `as_deref` here, because in the end I want `Option<&'a [PlainUpdatePathNode]>` not `&Option<Vec<PlainUpdatePathNode>>`.